### PR TITLE
chore(docs): Update Upgrade Standard doc

### PR DESCRIPTION
**Pull request by crazyuploader**
## Pull Request

---

### 📄 Summary
This PR adds support for foundry in upgrade standard doc

Doc: [Link](https://staging.signoz.io/docs/operate/migration/upgrade-standard/)

#### Issues closed by this PR
Closes https://github.com/SigNoz/growth-pod/issues/1290

---

### ✅ Change Type
_Select all that apply_

- [ ] ✨ Feature
- [ ] 🐛 Bug fix
- [ ] ♻️ Refactor
- [ ] 🛠️ Infra / Tooling
- [ ] 🧪 Test-only

---

---

- Tests added/updated:
- Manual verification:
- Edge cases covered:

---

- Blast radius:
- Potential regressions:
- Rollback plan:

---

| Field | Value |
|------|-------|
| Deployment Type | Cloud / OSS / Enterprise |
| Change Type | Feature / Bug Fix / Maintenance |
| Description | User-facing summary |

---

### 📋 Checklist
- [ ] Tests added or explicitly not required
- [ ] Manually tested
- [ ] Breaking changes documented
- [ ] Backward compatibility considered

---

## 👀 Notes for Reviewers

---

### CODE REVIEW

File: data/docs/operate/migration/upgrade-standard.mdx [hunk 1]
```diff
@@ -15,6 +15,131 @@ This guide covers how to upgrade your SigNoz instance across different deploymen
...
 
+## Upgrade with Foundry
+
+<Admonition type="info" title="What is Foundry?" defaultCollapsed="true">
+Foundry is an open-source CLI that runs your self-hosted observability stack as code. One declarative config provisions the infrastructure, installs the SigNoz backend, and sets up OpenTelemetry collection on bare metal, containers, or Kubernetes. Learn more about <a href="https://github.com/SigNoz/foundry" target="_blank" rel="noopener noreferrer nofollow">Foundry</a>.
+</Admonition>
+
+Foundry upgrades a Docker Compose, Docker Swarm, Kubernetes, or systemd deployment through the same loop. Update `foundryctl`, check your upgrade path, set the target version in `casting.yaml`, then re-run `foundryctl cast`.
+
+<Admonition type="info" title="Render, Coolify, and Railway castings">
```
Comment (Nageshbansal) [inline data/docs/operate/migration/upgrade-standard.mdx:26]: Please remove this

File: data/docs/operate/migration/upgrade-standard.mdx [hunk 2]
```diff
@@ -15,6 +15,131 @@ This guide covers how to upgrade your SigNoz instance across different deploymen
...
+
+### Step 2: Check your upgrade path
+
+Some releases are mandatory stops that carry their own steps, and skipping one breaks schema or dashboard migrations. Run your current version through the [Upgrade Path Tool](https://signoz.io/upgrade-path/) and note every stop it lists.
+
+Upgrade to one stop at a time, in the order the tool returns. Read that release's guide, then run Step 3 and Step 4 with the stop as your target version. Repeat until you reach the version you want.
+
+### Step 3: Set the target version in casting.yaml
+
+Pin the SigNoz and collector images to the release you want. An unpinned component defaults to the `latest` tag, which makes an upgrade hard to repeat and hard to roll back.
```
Comment (Nageshbansal) [inline data/docs/operate/migration/upgrade-standard.mdx:46]:
```suggestion
Pin the SigNoz and collector images to the release you want. An unpinned component defaults to the `latest` tag.
```

File: data/docs/operate/migration/upgrade-standard.mdx [hunk 3]
```diff
@@ -15,6 +15,131 @@ This guide covers how to upgrade your SigNoz instance across different deploymen
...
+    spec:
+      image: signoz/signoz:v0.138.0
+  ingester:
+    spec:
+      image: signoz/signoz-otel-collector:v0.144.9
+```
+
+SigNoz and the SigNoz OTel Collector use separate version series, so the two tags do not match. Take each tag from its own releases page: <a href="https://github.com/SigNoz/signoz/releases" target="_blank" rel="noopener noreferrer nofollow">SigNoz releases</a> and <a href="https://github.com/SigNoz/signoz-otel-collector/releases" target="_blank" rel="noopener noreferrer nofollow">SigNoz OTel Collector releases</a>. If you also pin ClickHouse, keep it on a version the collector supports. Foundry checks the combination and stops with the required version when it does not match.
+
+For a `mode: systemd` deployment, image tags do not apply. Foundry runs the binaries already installed on the host, so Step 4 replaces those instead.
```
Comment (Nageshbansal) [inline data/docs/operate/migration/upgrade-standard.mdx:60]: Move it to Linux install section/tab pls

File: data/docs/operate/migration/upgrade-standard.mdx [hunk 4]
```diff
@@ -15,6 +15,131 @@ This guide covers how to upgrade your SigNoz instance across different deploymen
...
+
+### Step 4: Apply the casting
+
+<Tabs entityName="environment">
+<TabItem value="containers" label="Docker, Swarm, and Kubernetes" default>
```
Comment (Nageshbansal) [inline data/docs/operate/migration/upgrade-standard.mdx:65]: Please split into two tabs: 
1. Docker Compose, Docker Swarm 
2. Kubernetes

File: data/docs/operate/migration/upgrade-standard.mdx [hunk 5]
```diff
@@ -15,6 +15,131 @@ This guide covers how to upgrade your SigNoz instance across different deploymen
...
+foundryctl cast -f casting.yaml
+```
+
+`cast` re-checks your tools, regenerates the files under `pours/`, and rolls out the new version:
+
+- Docker Compose: `docker compose up -d` against the regenerated Compose file
+- Docker Swarm: `docker stack deploy`
+- Kubernetes with Helm: `helm upgrade` on your existing release
+
+The telemetry-store migrator runs as part of the rollout, so schema migrations need no separate step.
```
Comment (Nageshbansal) [inline data/docs/operate/migration/upgrade-standard.mdx:77]: remove this

File: data/docs/operate/migration/upgrade-standard.mdx [hunk 6]
```diff
@@ -15,6 +15,131 @@ This guide covers how to upgrade your SigNoz instance across different deploymen
...
+
+<Admonition type="warning" title="On Docker Compose with unpinned images? Pull first">
```
Comment (Nageshbansal) [inline data/docs/operate/migration/upgrade-standard.mdx:79]: move it up before the cast command

File: data/docs/operate/migration/upgrade-standard.mdx [hunk 7]
```diff
@@ -15,6 +15,131 @@ This guide covers how to upgrade your SigNoz instance across different deploymen
...
+2. **Extract the new archives over the existing directories.** Use your own target versions, and your own paths if the casting annotations point elsewhere.
+   ```bash
+   ARCH=$(uname -m | sed 's/x86_64/amd64/;s/aarch64/arm64/')
+   curl -fsSL "https://github.com/SigNoz/signoz/releases/download/v0.138.0/signoz_linux_${ARCH}.tar.gz" \
+     | sudo tar -xz --strip-components=1 -C /opt/signoz
+   curl -fsSL "https://github.com/SigNoz/signoz-otel-collector/releases/download/v0.144.9/signoz-otel-collector_linux_${ARCH}.tar.gz" \
+     | sudo tar -xz --strip-components=1 -C /opt/ingester
+   ```
+
+   Extract each archive in full. The `signoz` binary loads the UI and the email templates relative to itself, so `bin/`, `web/`, `templates/`, and `conf/` must stay together.
```
Comment (Nageshbansal) [inline data/docs/operate/migration/upgrade-standard.mdx:107]: This wont be needed as per an upgrade
